src/ace/managers/blit.c

```diff
@@ -362,10 +360,10 @@ UBYTE blitUnsafeCopyMask(
 		g_pCustom->bltcmod = wDstModulo;
 		g_pCustom->bltdmod = wDstModulo;
 
-		g_pCustom->bltapt = (UBYTE*)((ULONG)pMsk + ulSrcOffs);
-		g_pCustom->bltbpt = (UBYTE*)((ULONG)pSrc->Planes[0] + ulSrcOffs);
-		g_pCustom->bltcpt = (UBYTE*)((ULONG)pDst->Planes[0] + ulDstOffs);
-		g_pCustom->bltdpt = (UBYTE*)((ULONG)pDst->Planes[0] + ulDstOffs);
+		g_pCustom->bltapt = pMsk + ulSrcOffs;
```

Comment:
whoops! pMsk is UWORD*, so adding hat offset directly won't work. I think it's safe to change that param to UBYTE*.